src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -125,9 +120,44 @@ create_distributed_table(PG_FUNCTION_ARGS)
 	text *distributionColumnText = PG_GETARG_TEXT_P(1);
 	Oid distributionMethodOid = PG_GETARG_OID(2);
 
+	text *colocateWithTableNameText = PG_GETARG_TEXT_P(3);
+	char *colocateWithTableName = text_to_cstring(colocateWithTableNameText);
+
 	char *distributionColumnName = text_to_cstring(distributionColumnText);
 	char distributionMethod = LookupDistributionMethod(distributionMethodOid);
 
+	/* check if we try to colocate with hash distributed tables */
+	if (strncmp(colocateWithTableName, "default", NAMEDATALEN) != 0 &&
+		strncmp(colocateWithTableName, "none", NAMEDATALEN) != 0)
+	{
+		bool colocateWithSupported = true;
+		Oid colocateWithTableOid = InvalidOid;
+		char colocateWithTablePartitionType = 0;
+
+		/* check the distribution type of the new table  */
+		if (distributionMethod != DISTRIBUTE_BY_HASH)
```

Comment:
I was a bit confused by this code at first, since I expected we would first try to parse the table name after confirming it's not either "none" or "default". 

What if we did something like `if (distributionMethod != DISTRIBUTE_BY_HASH || colocateWithTablePartitionType != DISTRIBUTE_BY_HASH) { ereport(...) }` below?